src/NuGet.Core/NuGet.CommandLine.XPlat/Commands/WhyCommand.cs

```diff
@@ -0,0 +1,107 @@
+// Copyright (c) .NET Foundation. All rights reserved.
+// Licensed under the Apache License, Version 2.0. See License.txt in the project root for license information.
+
+using System;
+using System.Globalization;
+using System.IO;
+using System.Linq;
+using Microsoft.Extensions.CommandLineUtils;
+using NuGet.Common;
+using NuGet.Frameworks;
+
+namespace NuGet.CommandLine.XPlat
+{
+    internal static class WhyCommand
+    {
+        public static void Register(CommandLineApplication app, Func<ILogger> getLogger, Func<IWhyCommandRunner> getCommandRunner)
```

Comment:
I am not sure if we are still working on it but there is a plan to migrate from `Microsoft.Extensions.CommandLineUtils.CommandLineApplication` to `System.CommandLine.CliCommand`. As a result, when we add new commands, it's recommended to use `System.CommandLine.CliCommand` instead. @martinrrm do you know if that is still the case?

Reply:
I agree, we should do System.Commandline.CLiCommand.

Reply:
I've changed this to CliCommand now. Please let me know if it doesn't look right or needs further changes